### PR TITLE
add code to get documents list by ticket with api

### PR DESCRIPTION
**Pull request by grandoc**

### CODE REVIEW

File: htdocs/api/class/api_documents.class.php [hunk 1]
```diff
@@ -472,6 +472,20 @@ public function getDocumentsListByElement($modulepart, $id = 0, $ref = '', $sort
 			}
 
 			$upload_dir = $conf->knowledgemanagement->dir_output.'/knowledgerecord/'.dol_sanitizeFileName($object->ref);
+		} elseif ($modulepart == 'ticket') {
+			require_once DOL_DOCUMENT_ROOT.'/ticket/class/ticket.class.php';
+
+			if (!DolibarrApiAccess::$user->rights->societe->client->voir && !DolibarrApiAccess::$user->rights->societe->client->voir) {
```
Comment (eldy) [inline htdocs/api/class/api_documents.class.php:478]: Permission seems the permission of another module.

File: htdocs/api/class/api_documents.class.php [hunk 2]
```diff
@@ -472,6 +472,20 @@ public function getDocumentsListByElement($modulepart, $id = 0, $ref = '', $sort
...
+				throw new RestException(401);
+			}
+
+			$object = new Ticket($this->db);
+			$result = $object->fetch($id, $ref);
+			if (!$result) {
+				throw new RestException(404, 'Expense report not found');
```
Comment (eldy) [inline htdocs/api/class/api_documents.class.php:485]: Expense report -> Ticket

File: htdocs/api/class/api_documents.class.php (hunk 1 above)
Comment (grandoc) [reply htdocs/api/class/api_documents.class.php:478]: into the ticket class all the rights are of that kind : $user->rights->societe->client->voir

Comment (eldy) [reply htdocs/api/class/api_documents.class.php:478]: No the use of $user->rights->societe->client->voir is only when we need a link with a sale representative of a thirdparty.
For ticket CRUD method, permissions must be the permission of module ticket so user->rights->ticket->...

Comment (grandoc) [reply htdocs/api/class/api_documents.class.php]: I made the rights changes

File: htdocs/api/class/api_documents.class.php [hunk 3]
```diff
@@ -472,6 +472,20 @@ public function getDocumentsListByElement($modulepart, $id = 0, $ref = '', $sort
 			}
 
 			$upload_dir = $conf->knowledgemanagement->dir_output.'/knowledgerecord/'.dol_sanitizeFileName($object->ref);
+		} elseif ($modulepart == 'ticket') {
+			require_once DOL_DOCUMENT_ROOT.'/ticket/class/ticket.class.php';
+
+			if (!DolibarrApiAccess::$user->rights->ticket->read && !DolibarrApiAccess::$user->rights->ticket->read) {
```
Comment (frederic34) [inline htdocs/api/class/api_documents.class.php:478]: @grandoc `!DolibarrApiAccess::$user->rights->ticket->read && !DolibarrApiAccess::$user->rights->ticket->read` there is twice the same right

**Comment by eldy** [comment]
CI error for too long, so i close PR

**Pull request closed without merging**